crates/volta-core/src/tool/package/install.rs

```diff
@@ -266,55 +278,45 @@ fn read_bins(name: &str, version: &Version) -> Fallible<HashMap<String, String>>
     Ok(bin_map)
 }
 
+/// Read the script for a shebang loader. If found, return it so it will be stored in the config
+///
+/// This is needed on Windows because Windows doesn't support shebang loaders for scripts
+/// On Unix, we need to do this to remove any potential erroneous \r characters that may
+/// have accidentally been published in the script.
+fn determine_script_loader(bin_name: &str, full_path: &Path) -> Fallible<Option<BinLoader>> {
+    let script =
+        File::open(full_path).with_context(|_| ErrorDetails::DetermineBinaryLoaderError {
+            bin: bin_name.into(),
+        })?;
+    if let Some(Ok(first_line)) = BufReader::new(script).lines().next() {
+        if let Some(caps) = SHEBANG.captures(&first_line) {
+            let args = caps["args"]
```

Comment:
If I'm remembering/understanding the docs right, this will always succeed because we only get here if we have captures, and the regex is constructed such that it *cannot* succeed without the presence of `args` in the string being targeted, right? (Just wanted to make sure I understood the invariant which guarantees this won't ever panic!)

Reply:
Oh, and now I see it's been around for a while, too!

Reply:
Right, the `Index` impl will only panic ["if there is no group named by the given value"](https://docs.rs/regex/1.3.3/regex/struct.Captures.html#panics-1). In this case, the named group is always part of the match (not optional or part of an "or" condition), so if there is a match, it will include the group.

I'll add a comment about that invariant, however, in case something changes in the future and we can't figure out why we did it that way :smiley: